backend/api/views/HeatView.py

```diff
@@ -0,0 +1,131 @@
+        try:
+            event_instance = MeetEvent.objects.get(id=event_id)
+        except MeetEvent.DoesNotExist:
+            return Response({'error': 'Event not found'}, status=status.HTTP_404_NOT_FOUND)
+        
+        #Get num_lanes
+        try:
+            swim_meet_instance = event_instance.swim_meet
+            num_lanes = swim_meet_instance.site.num_lanes
+        except:
+            return Response({'error': 'Not able to retrieve number of lanes'}, status=status.HTTP_404_NOT_FOUND)
+        
+        max_num_heat = Heat.objects.filter(event_id=event_id).aggregate(max_num_heat=Max('num_heat'))['max_num_heat']
+
+        if max_num_heat is not None:
+            if 1<= lane_num <= num_lanes:
+                # Retrieve all heats for the given event_id and heat_num
+                heats = Heat.objects.filter(event_id=event_id, lane_num=lane_num)
+                heats_data = []
+                for heat in range(1, max_num_heat + 1):
+                    heat_data = heats.filter(num_heat=heat).first()
+                    if heat_data is not None:
+                        heats_data.append(heat_data)
+                    else:
+                        heats_data.append({
+                            "id": None,
+                            "num_heat": heat,
+                            "athlete": None,
+                            "seed_time": None
+                    })
+        
+                # Serialize the heats using CompleteHeatSerializer
+                serializer = LaneSerializer(heats_data, many=True)
+        
+                return Response(serializer.data, status=status.HTTP_200_OK)
+            else:
+                return Response({'error': 'There is not a lane with that number'}, status=status.HTTP_404_NOT_FOUND)
+        else:
+            return Response({'message': 'This event does not have heats yet'}, status=status.HTTP_200_OK)
+
+    @transaction.atomic
+    def put(self, request, event_id, lane_num): 
+        # Given all the athletes on a lane, update the heat_time and register the time on TimeRecord.
+        try:
+            event_instance = MeetEvent.objects.get(id=event_id)
+        except MeetEvent.DoesNotExist:
+            return Response({'error': 'Event not found'}, status=status.HTTP_404_NOT_FOUND)
+        
+        #Get num_lanes
+        try:
+            swim_meet_instance = event_instance.swim_meet
+            num_lanes = swim_meet_instance.site.num_lanes
+        except:
+            return Response({'error': 'Not able to retrieve number of lanes'}, status=status.HTTP_404_NOT_FOUND)
+        
+        if 1<= lane_num <= num_lanes: 
+            result_data = request.data
+            for heat_record in result_data:
+                if heat_record['athlete'] is not None:
+                    num_heat = heat_record['num_heat']
+                    try:
+                        heat_instance = Heat.objects.get(event=event_instance,num_heat=num_heat, lane_num=lane_num)
```

Comment:
I think the athlete id should be added when getting the heat instance.

I sent a request with an incorrect athlete id and even thought the data was invalid, the record was updated for the heat with the num_heat and lane_num. i.e. I'm passing athlete id 11 (which is not correct) instead of id 12 (correct one), and the record time was updated for id 12.

<img width="1148" alt="Screenshot 2024-04-11 at 2 10 14 PM" src="https://github.com/MarisiaS/SMM/assets/102187795/216e4f39-e0b8-47d7-ac8f-9a93c5dcd376">

The similar way, I pasted in the request body the testing values provided for lane_num=5 but forgot to change the lane_num, I left 3 instead of update it to 5, and the record for lane_num=3, num_heat=2 (athlete id = 6) was updated to 25.43 even when the athlete id (4) was not the right one:

[
{
"num_heat": 1,
"athlete": null,
"heat_time":""
},
{
"num_heat": 2,
"athlete": 4,
"heat_time":"25.43"
}
]

<img width="1179" alt="Screenshot 2024-04-11 at 2 13 26 PM" src="https://github.com/MarisiaS/SMM/assets/102187795/5eb68637-aba8-411e-97de-48bd1e64bfc1">

Reply:
I made athlete and athlete_full_name read only. 
I think we need an endpoint to edit the athlete on lane, but the logic is more complicated. We have to think about that.

Reply:
As discussed, we will work on that in a separate Issue.